_docs/concepts/auth.md

```diff
@@ -0,0 +1,138 @@
+
+### Architecture {#architecture}
+
+Figure 1 shows the Istio Auth architecture, which includes three important components: identity, key management, and communication security. This diagram describes how Istio Auth is used to secure the service-to-service communication between service A running as the service account "foo" and service B running as the service account "bar".
+
+<img style="display:block;width:60%;margin:auto;" src="./img/auth/auth.svg"
+alt="Flow of traffic." />
+
+### Components {#components}
+
+#### Identity {#identity}
+
+Istio Auth uses [Kubernetes service accounts](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) to identify who runs the service:
+
+
+
+*   Service account is **the identity (or role) the workload runs as**, which represents the privilege of the workload. For the system requiring strong security, the amount of privilege for a workload should not be identified by a random string (i.e., service name, label, etc), or by the binary that is deployed.
+    *   For example, let's say we have a workload pulling data from a multi-tenant database. If Alice ran this workload, she will be able to pull a different set of data than if Bob ran this workload. 
+*   Service account enables powerful security policy by offering the flexibility to identify a machine, a user, a workload, or a group of workloads (different workloads can run as the same service account).
+*   The service account a workload runs as won't change during the lifetime of the workload.
+*   Service account uniqueness can be ensured with domain name constraint
+
+#### Communication Security {#communication-security}
+
+Service-to-service communication is tunneled through the client side [Envoy](https://lyft.github.io/envoy/) and the server side Envoy. The end-to-end communication is secured by:
+
+
+
+*   Local TCP connections between the service and Envoy
+    *   We are looking into using unix domain socket for stronger security
+*   Mutual TLS connections between proxies
+*   Secure Naming: during the handshake process, the client side Envoy checks that the service account provided by the server side certificate is allowed to run the target service
+
+#### Key Management {#key-management}
+
+Istio Auth provides a per-cluster CA (Certificate Authority) to automate key & cert management. It mainly performs 4 key operations:
+
+
+
+*   Generate a key/cert pair for each service account.
+*   Distribute key/cert to each pod according to the service account.
+*   Rotate key/cert periodically. 
+*   Revoke a specific key/cert pair when necessary.
+
+### Workflow 
+
+Istio Auth workflow consists of two phases, deployment and runtime. We briefly cover each phase in this section and a more detailed version can be found [here](https://docs.google.com/a/google.com/document/d/1spoQ9MIb7ABFDdFzlFITczCbH_AHO3RXSgLLeXAYIJU/edit?usp=sharing).
```

Comment:
This section briefly covers each phase - a more detailed version can be found...

[are we planning to publish the detailed version too? Random Google docs are kind of annoying as "real" documentation.]